src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.AbstractRecoverableSyntaxRoot.cs

```diff
@@ -161,19 +162,67 @@ protected override async Task<TRoot> RecoverAsync(CancellationToken cancellation
             {
                 Contract.ThrowIfNull(_storage);
 
-                using (var stream = await _storage.ReadStreamAsync(cancellationToken).ConfigureAwait(false))
+                var tickCount = Environment.TickCount;
+                try
                 {
-                    return RecoverRoot(stream, cancellationToken);
+                    if (RoslynEventSource.Instance.IsEnabled(EventLevel.Informational, EventKeywords.None))
+                    {
+                        RoslynEventSource.Instance.BlockStart(_containingTree.FilePath, FunctionId.Workspace_Recoverable_RecoverRootAsync, blockId: 0);
```

Comment:
Is there a reason to keep using `RoslynEventSource.Instance` vs. capturing it in a local?

Reply:
Not especially. I just used the pattern seen in similar features for other libraries we ship.